src/Apps/Settings/Routes/SavedSearchAlerts/SavedSearchAlertsApp.tsx

```diff
@@ -48,19 +50,35 @@ export const SavedSearchAlertsApp: React.FC<SavedSearchAlertsAppProps> = ({
   me,
   relay,
 }) => {
-  const [
-    editAlertEntity,
-    setEditAlertEntity,
-  ] = useState<EditAlertEntity | null>(null)
   const { relayEnvironment } = useSystemContext()
   const { sendToast } = useToasts()
   const { trackEvent } = useTracking()
+  const { silentPush } = useRouter()
   const [showDeleteModal, setShowDeleteModal] = useState(false)
   const [sort, setSort] = useState("ENABLED_AT_DESC")
   const [loading, setLoading] = useState(false)
   const alerts = extractNodes(me.alertsConnection)
+  const isMobile = __internal__useMatchMedia(THEME.mediaQueries.xs)
+
+  const [
+    editAlertEntity,
+    setEditAlertEntity,
+  ] = useState<EditAlertEntity | null>(null)
+
   const isEditMode = editAlertEntity !== null
-  const { match, silentPush } = useRouter()
+
+  useEffect(() => {
+    if (!alerts || !alerts[0]) return
+    if (isMobile === null) return
+    if (!isMobile) {
+      setEditAlertEntity({
+        id: alerts[0].internalID,
+        name: alerts[0].settings.name ?? "",
+        artistIds: alerts[0]?.artistIDs as string[],
+      })
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [isMobile])
```

Comment:
I feel this logic is quite complicated and it could use some testing. the reason behind it is that it adds some side effect that is not trivial and that will make debugging hard in case of an error. What I suggest to move on with this, is speaking to Barney about a robust design solution here. Something like:

<img width="1155" alt="Screenshot 2024-02-27 at 17 04 05" src="https://github.com/artsy/force/assets/11945712/2c5a4671-ecd8-4ea2-9b8f-64164d502ca9">

Reply:
@damassi I wonder how would you approach this? do you think this is worth the risk to build?

Reply:
I couldn't say, unfortunately, just too far away from the feature. 

If we do want to go ahead with the current implementation, I think updating the code to something like:

```tsx
const isMobile = getENV("IS_MOBILE") 

useEffect(() => {
  if (isMobile) {
    // do thing
  }
}, [isMobile])
```

Is fine, but this won't execute responsively, and that's ok -- a user can't drag around their browser window. 

So yah, that's my take, just from a code perspective!

Reply:
thanks, sorry for putting you on the spot 😸